lib/choria/orchestrator.rb

```diff
@@ -21,12 +21,15 @@ def tasks_support
       @tasks_support ||= MCollective::Util::Choria.new.tasks_support
     end
 
-    def run(task, targets: nil, verbose: false)
-      logger.debug "Running task: '#{task.name}' (targets: #{targets.nil? ? 'all' : targets})"
+    def run(task, targets: nil, targets_with_classes: nil, verbose: false)
       rpc_client.progress = verbose
 
+      logger.debug "Running task: '#{task.name}' (targets: #{targets.nil? ? 'all' : targets})"
       targets&.each { |target| rpc_client.identity_filter target }
 
+      logger.debug "Filtering targets with classes: #{targets_with_classes}" unless targets_with_classes.nil?
+      targets_with_classes&.each { |klass| rpc_client.class_filter klass }
```

Comment:
The `unless` on the logger line makes me think ([and should probably not](https://en.wikipedia.org/wiki/Don't_Make_Me_Think)), might be more readable as:

```ruby
if targets_with_classes
  logger.debug "Filtering targets with classes: #{targets_with_classes}"
  targets_with_classes.each { |klass| rpc_client.class_filter klass }
end
```

Reply:
OK, suggestion applied in `wip` branch.